modal/image.py

```diff
@@ -638,7 +638,14 @@ def add_local_file(self, local_path: Union[str, Path], remote_path: str, *, copy
         mount = _Mount.from_local_file(local_path, remote_path)
         return self._add_mount_layer_or_copy(mount, copy=copy)
 
-    def add_local_dir(self, local_path: Union[str, Path], remote_path: str, *, copy: bool = False) -> "_Image":
+    def add_local_dir(
+        self,
+        local_path: Union[str, Path],
+        remote_path: str,
+        *,
+        copy: bool = False,
+        include_files: Optional[Callable[[Path], bool]] = None,
```

Comment:
I was sort of imagining that we would support a string-based API for pattern based filtering, seems easier than asking users to construct their own callable. Also less flexible though! Would be fun to think through the different things users are going to want to do with this.

Reply:
Added this signature now: `ignore: Union[list[str], Callable[[Path], bool]] = []` to allow for both use cases!